pkg/antctl/raw/traceflow/command.go

```diff
@@ -0,0 +1,337 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traceflow
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/antctl/runtime"
+	"github.com/vmware-tanzu/antrea/pkg/apis/ops/v1alpha1"
+	clientset "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+)
+
+var (
+	Command     *cobra.Command
+	source      string
+	destination string
+	outputType  string
+	waiting     bool
+	flow        string
+)
+
+// Response is the response of antctl traceflow.
+type Response struct {
+	Name        string                  `json:"name" yaml:"name"`      // Traceflow name
+	Phase       v1alpha1.TraceflowPhase `json:"phase,omitempty"`       // Traceflow phase
+	Source      string                  `json:"source,omitempty"`      // Traceflow source, e.g. "default/pod0"
+	Destination string                  `json:"destination,omitempty"` // Traceflow destination, e.g. "default/pod1"
+	NodeResults []v1alpha1.NodeResult   `json:"results,omitempty"`     // Traceflow node results
+}
+
+func init() {
+	Command = &cobra.Command{
+		Use:     "traceflow",
+		Short:   "Start a Traceflows",
+		Long:    "Start a Traceflows from one pod to another.",
+		Aliases: []string{"tf", "traceflows"},
+		Example: `  Start a Traceflow from busybox0 to busybox1, both pods are in namespace default
+  $antctl traceflow -S busybox0 -D busybox1
+  Start a Traceflow from busybox0 to destination IP, source is in namespace default
+  $antctl traceflow -S busybox0 -D 123.123.123.123
+  Start a Traceflow from busybox0 in namespace ns0 to busybox1 in namespace ns1, output type is json
+  $antctl traceflow -S ns0/busybox0 -D ns1/busybox1 -o json
+  Start a Traceflow from busybox0 to busybox1, with TCP header and 80 as destination port
+  $antctl traceflow -S busybox0 -D busybox1 -f tcp,tcp_dst=80
+`,
+		RunE: runE,
+	}
+
+	Command.Flags().StringVarP(&source, "source", "S", "", "source of the traceflow: namespace/pod or pod")
+	Command.Flags().StringVarP(&destination, "destination", "D", "", "destination of the traceflow: namespace/pod, pod or IP")
+	Command.Flags().StringVarP(&outputType, "output", "o", "yaml", "output type: yaml (default), json")
+	Command.Flags().BoolVarP(&waiting, "wait", "", true, "if false, command returns without retrieving results")
```

Comment:
I don't know which case is this flag for because it looks there is no way to just retrieve the result without create CRD.

Reply:
If wait==false, command will only creates the traceflow.